integrations/appsflyer/src/main/kotlin/com/rudderstack/integration/kotlin/appsflyer/Utils.kt

```diff
@@ -0,0 +1,364 @@
+    ECommerceParamNames.ORDER_ID, ECommerceParamNames.SHARE_MESSAGE, CREATIVE,
+    ECommerceParamNames.RATING
+)
+
+private const val FIRST_PURCHASE = "first_purchase"
+private const val REMOVE_FROM_CART = "remove_from_cart"
+private const val AF_ORDER_ID = "af_order_id"
+private const val PRODUCT_ID = "product_id"
+private const val CATEGORY = "category"
+private const val QUANTITY = "quantity"
+
+/**
+ * Maps RudderStack event to AppsFlyer event name and properties
+ */
+internal fun mapEventToAppsFlyer(eventName: String, properties: JsonObject?): Pair<String, MutableMap<String, Any>> {
+    val appsFlyerEventProps = mutableMapOf<String, Any>()
+    val appsFlyerEventName = getAppsFlyerEventName(eventName, properties, appsFlyerEventProps)
+    return Pair(appsFlyerEventName, appsFlyerEventProps)
+}
+
+@Suppress("CyclomaticComplexMethod")
+private fun getAppsFlyerEventName(
+    eventName: String,
+    properties: JsonObject?,
+    appsFlyerEventProps: MutableMap<String, Any>
+): String {
+    return when (eventName) {
+        ECommerceEvents.PRODUCTS_SEARCHED -> handleProductsSearched(properties, appsFlyerEventProps)
+        ECommerceEvents.PRODUCT_VIEWED -> handleProductViewed(properties, appsFlyerEventProps)
+        ECommerceEvents.PRODUCT_LIST_VIEWED -> handleProductListViewed(properties, appsFlyerEventProps)
+        ECommerceEvents.PRODUCT_ADDED_TO_WISH_LIST -> handleProductWishList(properties, appsFlyerEventProps)
+        ECommerceEvents.PRODUCT_ADDED -> handleProductAdded(properties, appsFlyerEventProps)
+        ECommerceEvents.CHECKOUT_STARTED -> handleCheckoutStarted(properties, appsFlyerEventProps)
+        ECommerceEvents.ORDER_COMPLETED -> handleOrderCompleted(properties, appsFlyerEventProps)
+        FIRST_PURCHASE -> handleFirstPurchase(properties, appsFlyerEventProps)
+        ECommerceEvents.PRODUCT_REMOVED -> handleProductRemoved(properties, appsFlyerEventProps)
+        ECommerceEvents.PROMOTION_VIEWED -> handlePromotionViewed(properties, appsFlyerEventProps)
+        ECommerceEvents.PROMOTION_CLICKED -> handlePromotionClicked(properties, appsFlyerEventProps)
+        ECommerceEvents.PAYMENT_INFO_ENTERED -> AFInAppEventType.ADD_PAYMENT_INFO
+        ECommerceEvents.PRODUCT_SHARED, ECommerceEvents.CART_SHARED -> handleShared(properties, appsFlyerEventProps)
+        ECommerceEvents.PRODUCT_REVIEWED -> handleProductReviewed(properties, appsFlyerEventProps)
+        else -> eventName.lowercase().replace(" ", "_")
+    }
+}
+
+private fun handleProductsSearched(properties: JsonObject?, appsFlyerEventProps: MutableMap<String, Any>): String {
+    properties?.getTypedValue(ECommerceParamNames.QUERY)?.let { query ->
+        appsFlyerEventProps[AFInAppEventParameterName.SEARCH_STRING] = query
+    }
+    return AFInAppEventType.SEARCH
+}
+
+private fun handleProductViewed(properties: JsonObject?, appsFlyerEventProps: MutableMap<String, Any>): String {
```

Comment:
Can you give a thought if we could combine all these methods into one:

```
private fun handleProductsSearched(properties: JsonObject?, appsFlyerEventProps: MutableMap<String, Any>): String {

private fun handleProductViewed(properties: JsonObject?, appsFlyerEventProps: MutableMap<String, Any>): String {

private fun handleProductListViewed(properties: JsonObject?, appsFlyerEventProps: MutableMap<String, Any>): String {

// ...
```

They seem to have the same set of parameter list.

Reply:
If the current structure looks good, feel free to proceed with it.